components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,203 @@
+	}
+
+	for i := 0; i < len(cmdArray); i++ {
+		checkResponse := sv.checkCommandVersion(cmdArray[i])
+		if !checkResponse.Passed {
+			serviceResponse.Passed = false
+		}
+		serviceResponseArray = append(serviceResponseArray, *checkResponse)
+	}
+	osResponse, err := sv.checkOsVersion(sv.osFilepath)
+	if err != nil {
+		sv.logger.Error("Error while getting the OS Version = ", err)
+	}
+	if !osResponse.Passed {
+		serviceResponse.Passed = false
+	}
+	serviceResponseArray = append(serviceResponseArray, *osResponse)
+	serviceResponse.Checks = serviceResponseArray
+	sv.logger.Debug("The Passed value for the response = ", serviceResponse.Passed)
+	sv.logger.Debug("The Checks array for the response = ", serviceResponse.Checks)
+	return &models.SoftwareVersionDetails{
+		Passed: serviceResponse.Passed,
+		Checks: serviceResponse.Checks,
+	}, nil
+}
+
+func (sv *SoftwareVersionService) checkCommandVersion(cmdName string) *models.Checks {
+	_, err := fiberutils.CheckPath(cmdName)
+	sv.logger.Error("The errror which checking cammand file path = ",err)
+	if err != nil {
+		return &models.Checks{
+			Title:         cmdName + AVAILABILITY,
+			Passed:        false,
+			SuccessMsg:    "",
+			ErrorMsg:      cmdName + " is not available",
+			ResolutionMsg: ENSURE + cmdName + " is available in $PATH on the node",
+		}
+	}
+	return &models.Checks{
+		Title:         cmdName + AVAILABILITY,
+		Passed:        true,
+		SuccessMsg:    cmdName + " is available",
+		ErrorMsg:      "",
+		ResolutionMsg: "",
+	}
+}
+
+func (sv *SoftwareVersionService) checkOsVersion(osFilepath string) (*models.Checks, error) {
+	osVersions := map[string][]string{
+		"Red Hat":      {"7,8,9"},
+		"Ubuntu":       {"16.04.x", "18.04.x", "20.04.x", "22.04.x"},
+		"Centos":       {"7"},
+		"Amazon Linux": {"2"},
+		"SUSE Linux":   {"12"},
+		"Debian":       {"9", "10", "11", "12"},
+	}
+	checkResponse := models.Checks{}
+	var osName, osVersion, err = getosutils.GetOsVersion(osFilepath)
+	sv.logger.Debug("Got the OS Version = ", osVersion)
+	sv.logger.Debug("Got the OS Name = ", osName)
+	if err != nil {
+		sv.logger.Error("Enable to get OS Version as file on path doesnot exit = ", err)
+		return &models.Checks{
+			Title:         LINUX_VERSION_CHECK,
+			Passed:        false,
+			SuccessMsg:    "",
+			ErrorMsg:      "Its not feasible to determine the Operating system version",
+			ResolutionMsg: "Please run system on the supported platforms.",
+		}, nil
+	}
+	for key := range osVersions {
+		if strings.Contains(osName, key) {
+			correctVersion := sv.checkOs(osVersions, osVersion, key)
+			if correctVersion {
+				checkResponse = models.Checks{
+					Title:         LINUX_VERSION_CHECK,
+					Passed:        true,
+					SuccessMsg:    key + " version is " + osVersion,
+					ErrorMsg:      "",
+					ResolutionMsg: "",
+				}
+				break
+			}
+			checkResponse = models.Checks{
+				Title:         LINUX_VERSION_CHECK,
+				Passed:        false,
+				SuccessMsg:    "",
```

Comment:
since these fields are common you can declare the struct with these values prior and just do `checkResponse. ErrorMsg = "----"` where you need to add.
This will reduce few lines. Also if there are any other places you can do the same